app/models/user.rb

```diff
@@ -1,11 +1,20 @@
 class User < ActiveRecord::Base
   has_secure_password validations: false
-  has_many :reviews
+  has_many :reviews, -> { order("created_at DESC") }
   has_many :queue_items, -> { order(:list_order) }
+  has_many :following_relationships, class_name: 'Relationship', foreign_key: :follower_id
+  has_many :leading_relationships, class_name: 'Relationship', foreign_key: :leader_id
 
   validates_presence_of :email, :password, :full_name
   validates_uniqueness_of :email
 
+  def can_follow?(another_user)
+    self != another_user && !follows?(another_user)
+  end
+
+  def follows?(another_user)
+    following_relationships.map(&:leader).include?(another_user)
```

Comment:
Fun Fact (and something to be aware of) `map` pulls all of the objects which are found into memory and instantiates them. For that reason, it would be better to do this another way. For example, you could use `pluck`, which will allow you to grab just one column:

```ruby
following_relationships.pluck(:leader_id).include?(another_user.id)
```

You could also do:

```ruby
following_relationships.find_by(leader: another_user)
```

optionally you can add a double bang to the beginning of that one to make it a true boolean. One more option that comes to mind is:

```ruby
following_relationships.where(leader: another_user).any?
```